File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/entity/modifier/PositionModifier.java
/**
 * 
 */
package com.abubusoft.xenon.entity.modifier;

import com.abubusoft.xenon.entity.BaseEntity;

/**
 * Modificatore di posizione per le entità.
 * 
 * @author Francesco Benincasa
 *
 */
public class PositionModifier extends StaticModifier {
	
	public static void moveBy(BaseEntity entity, float xOffset, float yOffset, float zOffset)
	{
		entity.position.add(xOffset, yOffset, zOffset);
	}
	
	public static void moveBy(BaseEntity entity, float xOffset, float yOffset)
	{
		entity.position.add(xOffset, yOffset, 0);
	}
	
	public static void moveTo(BaseEntity entity, float x, float y)
	{
		entity.position.setCoords(x, y, 0);
	}
	
	public static void moveTo(BaseEntity entity, float x, float y, float z)
	{
		entity.position.setCoords(x, y, z);
	}
	
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/entity/modifier/PositionModifier.kt
/**
 *
 */
package com.abubusoft.xenon.entity.modifier

import com.abubusoft.xenon.entity.BaseEntity

/**
 * Modificatore di posizione per le entità.
 *
 * @author Francesco Benincasa
 */
object PositionModifier : StaticModifier() {
    fun moveBy(entity: BaseEntity, xOffset: Float, yOffset: Float, zOffset: Float) {
        entity.position.add(xOffset, yOffset, zOffset)
    }

    fun moveBy(entity: BaseEntity, xOffset: Float, yOffset: Float) {
        entity.position.add(xOffset, yOffset, 0f)
    }

    fun moveTo(entity: BaseEntity, x: Float, y: Float) {
        entity.position.setCoords(x, y, 0f)
    }

    fun moveTo(entity: BaseEntity, x: Float, y: Float, z: Float) {
        entity.position.setCoords(x, y, z)
    }
}